jena-arq/src/main/java/org/apache/jena/sparql/core/Transactional.java

```diff
@@ -31,32 +33,12 @@
  * <pre> begin(WRITE) ... abort() or commit()</pre>
  * <p>{@code end()} is optional but preferred.
  * <p>
- * Helper code is available {@link Txn} so, for example:
- * <pre>Txn.executeRead(dataset, {@literal ()->} { ... sparql query ... });</pre>
- * <pre>Txn.executeWrite(dataset, {@literal ()->} { ... sparql update ... });</pre>
- * or use one of <tt>Txn.calculateRead</tt> and <tt>Txn.executeWrite</tt>
+ * In most cases instead of calling primitive methods such as {@code begin} and {@code commit }
```

Comment:
"In most cases" begs the question of when is it appropriate to use begin-commit.

There will be application already existing that correctly uses begin-try-commit-finally and works just fine and we are keeping valid.

How about:

"Applications can call [examples e.g. dataset.executeRead] or use the lower-level primitives [dco for begin-commit].

By the way Txn will handle some nesting cases - no nested transactions but to cope with the case of an existing transaction.